eagle-xl/scripts/metallicity_evolution.py

```diff
@@ -0,0 +1,124 @@
+"""
+Plots the gas and stellar metallicity density evolution.
+"""
+import matplotlib
+
+import unyt
+
+import matplotlib.pyplot as plt
+import numpy as np
+import sys
+
+
+from swiftsimio import load, load_statistics
+
+from swiftpipeline.argumentparser import ScriptArgumentParser
+
+arguments = ScriptArgumentParser(
+    description="Creates a metallicity density evolution plot for gas, stars and black holes."
+)
+
+snapshot_filenames = [
+    f"{directory}/{snapshot}"
+    for directory, snapshot in zip(arguments.directory_list, arguments.snapshot_list)
+]
+
+stats_filenames = [
+    f"{directory}/statistics.txt" for directory in arguments.directory_list
+]
+
+names = arguments.name_list
+output_path = arguments.output_directory
+
+plt.style.use(arguments.stylesheet_location)
+
+simulation_lines = []
+simulation_labels = []
+
+fig, ax = plt.subplots()
+
+ax.loglog()
+
+for color, (snapshot_filename, stats_filename, name) in enumerate(
+    zip(snapshot_filenames, stats_filenames, names)
+):
+    data = load_statistics(stats_filename)
+
+    snapshot = load(snapshot_filename)
+    boxsize = snapshot.metadata.boxsize
+    box_volume = boxsize[0] * boxsize[1] * boxsize[2]
+
+    # a, Redshift, SFR
+    scale_factor = data.a
+    redshift = data.z
+    gas_Z_mass = data.gas_z_mass
+    star_Z_mass = data.star_z_mass
+    gas_Z_mass_density = gas_Z_mass / box_volume
+    star_Z_mass_density = star_Z_mass / box_volume
+
+    # High z-order as we always want these to be on top of the observations
+    simulation_lines.append(
+        ax.plot(
+            scale_factor,
+            gas_Z_mass_density,
+            linestyle="solid",
+            color=f"C{color}",
+            zorder=10000,
+        )[0]
+    )
+    simulation_lines.append(
+        ax.plot(
+            scale_factor,
+            star_Z_mass_density,
+            linestyle="dashed",
+            color=f"C{color}",
+            zorder=10000,
+        )[0]
+    )
+    simulation_labels.append(name)
+
+ax.set_xlabel("Redshift $z$")
+ax.set_ylabel(r"Metal Mass $\rho_{\rm Z}$ [M$_\odot$ Mpc$^{-3}$]")
```

Comment:
Again requires use of the unit conversions. Alternatively, you can use the `unyt.matplotlib_support` environment (see https://unyt.readthedocs.io/en/stable/usage.html#plotting-with-matplotlib)